aws/defaults/defaults.go

```diff
@@ -96,23 +97,46 @@ func CredChain(cfg *aws.Config, handlers request.Handlers) *credentials.Credenti
 	})
 }
 
+const (
+	httpProviderEnvVar     = "AWS_CONTAINER_CREDENTIALS_FULL_URI"
+	ecsCredsProviderEnvVar = "AWS_CONTAINER_CREDENTIALS_RELATIVE_URI"
+)
+
 // RemoteCredProvider returns a credenitials provider for the default remote
 // endpoints such as EC2 or ECS Roles.
 func RemoteCredProvider(cfg aws.Config, handlers request.Handlers) credentials.Provider {
-	ecsCredURI := os.Getenv("AWS_CONTAINER_CREDENTIALS_RELATIVE_URI")
+	if u := os.Getenv(httpProviderEnvVar); len(u) > 0 {
+		parsed, err := url.Parse(u)
+		// Need to log error because this API did not expose an error
+		if err != nil {
+			log(cfg.Logger,
+				"Ignoring,", httpProviderEnvVar, "failed to parse url", err)
```

Comment:
Hrm, so if it fails to parse, it'll still continue on. I wonder if it is better to just return and empty `Provider` or `nil`.